src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1766,3 +1760,23 @@ def list_locations(cmd, sku, linux_workers_enabled=None):
     client = web_client_factory(cmd.cli_ctx)
     full_sku = get_sku_name(sku)
     return client.list_geo_regions(full_sku, linux_workers_enabled)
+
+
+def _check_zip_deployment_status(deployment_status_url, authorization):
+    import requests
+    num_trials = 1
+    while num_trials < 200:
+        response = requests.get(deployment_status_url, headers=authorization)
+        res_dict = response.json()
+        num_trials = num_trials + 1
+        if res_dict['status'] == 5:
+            logger.warning("Zip deployment failed status %s", res_dict['status_text'])
+            break
+        elif res_dict['status'] == 4:
+            break
+        logger.warning(res_dict['progress'])
+    # if the deployment is taking longer than expected
+    if res_dict['status'] != 4:
+        logger.warning("""Deployment is taking longer than expected. Please verify status at '%s'
```

Comment:
use `'` to avoid double quotes

Reply:
@yugangw-msft to clarify i was using triple quotes """ for multi-line messages, is there another recommended approach for this?

Reply:
because it is in the parentheses, you don't need `"""`, I meant the following code should work.  Up to you though
```python
logger.warning("Deployment is taking longer than expected. Please verify status at '%s' "
               "beforing launching the app", deployment_status_url)
```